docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityBase.php

```diff
@@ -116,4 +140,216 @@ protected function isLovellTricareSection(EntityInterface $entity) : bool {
+    $hours->friday = $this->getDay(4, $office_hours);
+    $hours->saturday = $this->getDay(5, $office_hours);
+    $hours->sunday = $this->getDay(6, $office_hours);
+
+    return $hours;
+  }
+
+  /**
+   * Gets the string for a given day.
+   *
+   * @param int $day_num
+   *   Number 0-6 specifying which day to return.
+   * @param array $days
+   *   The array of days data.
+   *
+   * @return string
+   *   a sting made up of start and end times with comment, or just comment.
+   */
+  protected function getDay($day_num, array $days): string {
+    $start = (!empty($days[$day_num]['starthours'])) ? OfficeHoursDateHelper::format($days[$day_num]['starthours'], 'h:i a') : '';
+    $end = (!empty($days[$day_num]['endhours'])) ? OfficeHoursDateHelper::format($days[$day_num]['endhours'], 'h:i a') : '';
+    // Make sure there is no end if there is no start. Data error.
+    $end = empty($start) ? '' : $end;
+    $start = $this->normalizeTime($start);
+    $end = $this->normalizeTime($end);
+    $comment = $days[$day_num]['comment'] ?? '';
+    $comment = $this->normalizeHoursComment($comment);
+    if (!empty($start) && !empty($end) && !empty($comment)) {
+      $day_entry = "{$start} to {$end} {$comment}";
+    }
+    elseif (!empty($start) && empty($end) && !empty($comment)) {
+      $day_entry = "{$start} to {$comment}";
+    }
+    elseif (!empty($start) && !empty($end) && empty($comment)) {
+      $day_entry = "{$start} to {$end}";
+    }
+    else {
+      $day_entry = $comment;
+    }
+
+    return trim($day_entry);
+  }
+
+  /**
+   * Perform processes on a comment to make it normal.
+   *
+   * @param string $comment
+   *   A comment to normalize.
+   *
+   * @return string
+   *   A normalized comment.
+   */
+  protected function normalizeHoursComment(string $comment): string {
+    $comment = trim($comment);
+    // There will be more processes coming here.
+    return $comment;
+  }
+
+  /**
+   * Perform alterations to a time string to make normal.
+   *
+   * @param string $time
+   *   A formatted time string.
+   *
+   * @return string
+   *   The normalized string.
+   */
+  protected function normalizeTime(string $time): string {
+    $time = trim($time);
+    // Make am pm follow design.va.gov.
+    $replacements = [
+      'am' => 'a.m.',
+      'AM' => 'a.m.',
+      'A.M.' => 'a.m.',
+      'pm' => 'p.m.',
+      'PM' => 'p.m.',
+      'P.M.' => 'p.m.',
+    ];
+    $time = str_replace(array_keys($replacements), array_values($replacements), $time);
+    $time = $this->midnightNoonify($time);
+    return $time;
+  }
+
+  /**
+   * Convert noon or midnight to those terms.
+   *
+   * @param string $time
+   *   A time including am or pm.
+   *
+   * @return string
+   *   The time passed in or 'noon' or 'midnight'.
+   */
+  protected function midnightNoonify(string $time): string {
+    $time = trim($time);
+    if ($time === '12:00 am' || $time === '12:00 a.m.') {
+      $time = 'noon';
```

Comment:
I think these are backwards.
```suggestion
      $time = 'midnight';
```

Reply:
Great googly moogly.   Good catch.